unic/char/range/src/range.rs

```diff
@@ -148,6 +149,24 @@ impl CharRange {
         self.low <= ch && ch <= self.high
     }
 
+    /// Determine the ordering of this range and a character.
+    ///
+    /// # Panics
+    ///
+    /// Panics at runtime _in debug builds only_ if the range is empty.
+    /// In release builds, the ordering will never be `Equal`,
```

Comment:
IIUC, this claim is based on how this function is used, by not how it's implemented. I think it needs to be an assertion in some other part of the code, of in another function. From the API perspective, this function can be called with a `ch` value that falls properly in a non-empty range, correctly expecting `Equal` to be return. Or did I misunderstand this?

Reply:
This specifically is talking about the empty-range case. How could I make this clearer?

Reply:
Hum... but doesn't look like the language matches the code, for the case of release mode.

For example, if we have `chars!('z'..='a')`, this is an empty range, but comparing it to `'b'` will return `Ordering::Less` and not `Ordering::Equal`.

I think we need a more consistent behavior for empty ranges here. Since the signature doesn't allow to err, I guess it's better to just panic, in both debug and release.

What do you think?

Reply:
It's definitely simpler to panic. In any case, since we panic in debug mode you have to avoid it anyway.

But I've intentionally not returned `Ordering::Equal` -- `::cmp(ch) == Ordering::Equal` should be equivalent to `::contains(ch)`. That's what the docs say -- that the ordering is never `Equal` for an empty range, but rather an arbitrary value which is not `Equal`. If that's not clear, the docs should be improved.

I haven't benchmarked it, but since we have to call this so much when binary searching through a table, I want to keep it `debug_assert!` if at all possible. But I think I'll move this to always panic for the time being; we can do that optimization later if needed. (I'll make a note of the possibility.)